Lib/traceback.py

```diff
@@ -1520,8 +1520,11 @@ def _compute_suggestion_error(exc_value, tb, wrong_name):
         # has the wrong name as attribute
         if 'self' in frame.f_locals:
             self = frame.f_locals['self']
-            if hasattr(self, wrong_name):
-                return f"self.{wrong_name}"
+            try:
+                if hasattr(self, wrong_name):
+                    return f"self.{wrong_name}"
+            except Exception:
+                return None
```

Comment:
Now this close follows to the proposal in the issue thread, but the exception class is too broad.

```suggestion
            except NameError:
                return
```

Reply:
I wanted to handle broad exception like it was done in the upper if blocks.